suricata/update/net.py

```diff
@@ -111,25 +122,38 @@ def get(url, fileobj, progress_hook=None):
     else:
         opener.addheaders = [(header, value) for header,
                              value in opener.addheaders if header.lower() != "user-agent"]
-    remote = opener.open(url)
-    info = remote.info()
+    if isinstance(url, tuple):
+        header = url[1].split(":") if url[1] is not None else None
+        if header and is_header_clean(header=header):
+            name, val = header[0].strip(), header[1].strip()
+            logger.debug("Setting HTTP header %s to %s", name, val)
+            opener.addheaders = [(name, val), ]
```

Comment:
Just noticed this after some more testing - this appears to erase the user header set above.  Probably need to construct the list and then set it once before making the request.